### PR TITLE
Add X-TraceId

### PR DESCRIPTION
**Pull request by bobjflong**

### CODE REVIEW

File: shelduck.cabal [hunk 1]
```diff
@@ -37,6 +37,7 @@ library
                        , transformers
                        , shelly
                        , uuid
+                       , time
```
Comment (bobjflong) [inline shelduck.cabal:40]: gives us the `getPOSIXTime` function

File: shell.nix [hunk 2]
```diff
@@ -6,7 +6,7 @@ let
 
   f = { mkDerivation, aeson, async, base, bytestring, hastache
       , hspec, lens, lens-aeson, rainbow, random, regex-compat, shelly
-      , Spock, stdenv, stm, text, transformers, uuid, wreq
+      , Spock, stdenv, stm, text, time, transformers, uuid, wreq
```
Comment (bobjflong) [inline shell.nix:9]: this shell.nix is for nix users. Ignore :)

File: src/Shelduck/IntercomDefinitions.hs [hunk 3]
```diff
@@ -68,57 +75,68 @@ tagName = "foo"
...
 run t = void $ do
   threadDelay 5000000
+
   options <- opts
   contactResp <- go $ blank & requestEndpoint .~ "https://api.intercom.io/contacts"
                             & requestOpts .~ options
                             & requestParameters .~ object []
                             & requestTopic .~ "contact.created"
 
+  options <- opts
```
Comment (bobjflong) [inline src/Shelduck/IntercomDefinitions.hs:85]: generate a new `options` for every request

**Comment by bscanlan** [comment]
:bird:

**Pull request merged**